app/controllers/main/routes.py

```diff
@@ -37,38 +40,72 @@ def events(selectedTerm=None):
         currentTime = currentTime,
         user = g.current_user)
 
-@main_bp.route('/profile/<username>', methods = ['GET'])
-def profilePage(username):
-    if not g.current_user.isCeltsAdmin and not g.current_user.isCeltsStudentStaff and g.current_user.username != username:
-        return "Access Denied", 403
+@main_bp.route('/profile/<username>', methods=['GET'])
+def viewVolunteersProfile(username):
+    """
+    This function displays the information of a volunteer to the user
+    """
+    if (g.current_user.username == username) or g.current_user.isAdmin:
+         upcomingEvents = getUpcomingEventsForUser(username)
+         programs = Program.select()
+         interests = Interest.select().where(Interest.user == username)
+         programBan = ProgramBan.select().where(ProgramBan.user == username)
+         interests_ids = [interest.program for interest in interests]
+         eventParticipant = EventParticipant.select().where(EventParticipant.user == username)
+         currentTerm = Term.select().where(Term.isCurrentTerm == 1)
+         trainingEvents = (ProgramEvent.select().join(Event).where((Event.term == currentTerm) & (Event.isTraining == 1)))
+         trainingChecklist = {}
+         for program in programs:
+             trainingChecklist[program.id] = trainedParticipants(program.id)
+         eligibilityTable = []
+         for i in programs:
+             eligibilityTable.append({"program" : i,
+                                      "completedTraining" : (username in trainedParticipants(i)),
+                                      "isNotBanned" : isEligibleForProgram(i, username)})
+         return render_template ("/main/volunteerProfile.html",
+            programs = programs,
+            eventParticipant = eventParticipant,
+            interests = interests,
+            trainingEvents = trainingEvents,
+            programBan = programBan,
+            interests_ids = interests_ids,
+            trainingChecklist = trainingChecklist,
+            upcomingEvents = upcomingEvents,
+            eligibilityTable = eligibilityTable,
+            volunteer = User.get(User.username == username),
+            user = g.current_user)
+    abort(403)
+
+
+@main_bp.route('/banUnbanUser/<program_id>/<username>', methods=['POST'])
```

Comment:
Instead of having banUnbanUser, make two routes. /banUser and /unbanUser. Each route is a specific action, though they can use the same controller method, then you don't have to pass banOrUnban either.

Reply:
Two routes have been made.